content/en/docs/Writing policies/verify-images.md

```diff
@@ -4,4 +4,145 @@ description: Check image signatures and add digests
 weight: 3
 ---
 
-TBD...
\ No newline at end of file
+{{% alert title="Note" color="warning" %}}
+Image verification is an **preview** feature. It is not ready for production usage and will likely change. 
+{{% /alert %}}
+
+[Sigstore](https://sigstore.dev/) is a [Linux Foundation project](https://linuxfoundation.org/) focused on software signing and transparency log technologies to improve software supply chain security. [Cosign](https://github.com/sigstore/cosign) is a sub-project that provides image signing, verification, and storage in an OCI registry.
+
+The Kyverno `verifyImages` rule uses [Cosign](https://github.com/sigstore/cosign) to verify container image signatures stored in an OCI registry. The rule matches an image reference (wildcards are supported) and specifies a public key to be used to verify the signed image. The policy rule check fails if the image signature is not found in the OCI registry, or if the image was not signed using the specified key.
+
+The rule also mutates matching images to add the `image digest` if a digest is not already specified. Using an image digest has the benefit of making images references immutable. This helps ensure that the expected version of the image is being run, for example the exact version that was scanned and verified by a vulnerability detection tool.
```

Comment:
..."image references immutable."

Reply:
"This helps ensure that the expected version of the image that is being run, for example, is the exact version that was scanned and verified..."